src/compiler/checker.ts

```diff
@@ -4992,6 +4971,17 @@ namespace ts {
             return type;
         }
 
+        function cloneTypeReference(source: TypeReference): TypeReference {

```

Comment:
It clones any type reference, regardless of what it is a reference to. There's no particular use for that currently, but also no reason to special case for tuples.